utils.go

```diff
@@ -61,3 +61,43 @@ func DecodeLabel(image imgutil.Image, label string, v interface{}) error {
 	}
 	return nil
 }
+
+// SyncLabels adds/updates/removes labels matching the predicate and a source image
+func SyncLabels(sourceImg imgutil.Image, destImage imgutil.Image, test func(string) bool) error {
+	if err := RemoveLabels(destImage, test); err != nil {
+		return err
+	}
+	return CopyLabels(sourceImg, destImage, test)
+}
+
+// RemoveLabels removes labels from an image matching the predicate
+func RemoveLabels(image imgutil.Image, test func(string) bool) error {
+	labels, err := image.Labels()
+	if err != nil {
+		return err
+	}
+
+	for label := range labels {
+		if test(label) {
+			delete(labels, label)
```

Comment:
Thank you for making this change.
Two additional comments:
1. I think it's better that there will be a `RemoveLabel` function (like `SetLabel` on line 97) instead of calling the `delete` function directly in order to hide the implementation of `Labels`.
2. I'm not sure how golang behaves if you delete a `label` for `labels` during a for loop. I'm not sure if it's "smart" enough or we should create a list of "toDelete" labels during the for loop and then call the `RemoveLabel` on each one of them.

Reply:
2. In go, it is safe to remove in the loop. [Docs](https://golang.org/ref/spec#For_statements)

As someone who came from C#/Ruby, this feels weird but is totally OK :) 

I'll work on getting a PR up for `imgutil` to add the corresponding `RemoveLabel`. That is the only reason it wasn't done here, the change needs to be done in `imgutil`.

Reply:
I don't think this implementation of `RemoveLabels` will actually work for a `remote.Image`. When you call `Labels` on the fake or local image you get a pointer to the actual labels map that will be used to create the image config during save. However, the remote image is backed by a `ggcr` image and the config must be explicitly mutated (Example: https://github.com/buildpacks/imgutil/blob/main/remote/remote.go#L277).

We could make make the behavior of the fake and local image consistent with remote (in a way that would catch errors like this) by returning a copy of the `Labels` map from `image.Labels()` instead of returning the exact map used by the image.

Reply:
@ekcasey, maybe I missed something but are you saying that we can't add `RemoveLabel` to the `Image` interface?

Reply:
@yaelharel see https://github.com/buildpacks/imgutil/pull/61 - the implementation is specific to local/remote..so I think Emily is saying returning the actual map for `Labels()` for local/fake images could produce code that relies on that, which would fail on a remote image.

Reply:
@jabrown85 Yep that is what I meant. Thanks for restating it more clearly.

I think this code as originally PR'ed (without incorporating https://github.com/buildpacks/imgutil/pull/61) falls into that trap. Tests that assert on label removal pass here with the fake image, but would probably not pass with a remote image.

Reply:
Thank you both! Now it's much clearer!